### PR TITLE
Add min and max duration filters

### PR DESCRIPTION
**Pull request by Th0masL**
This PR allows to use `min_duration` and `max_duration` (in seconds) as filter to detect if we want to download the episode or not.

Example of log lines :
```
INFO[2022-11-19T22:50:49+02:00] skipping due to regexp mismatch               episode_id=NIMLW2Ruxyz filter=title
INFO[2022-11-19T22:50:49+02:00] skipping due to duration filter               episode_id=Q5a81_JNxyz filter=min_duration
DEBU[2022-11-19T22:50:49+02:00] adding f6Vkhov_xyz ("Some episode name") to queue
INFO[2022-11-19T22:50:49+02:00] download count: 1                            
INFO[2022-11-19T22:50:49+02:00] ! downloading episode https://youtube.com/watch?v=f6Vkhov_xyz  episode_id=f6Vkhov_xyz index=0

```

### CODE REVIEW

File: services/update/matcher.go [hunk 1]
```diff
@@ -41,5 +54,13 @@ func matchFilters(episode *model.Episode, filters *feed.Filters) bool {
 		return false
 	}
 
+	if !matchDurationFilter(filters.MinDuration, episode.Duration, "min", logger.WithField("filter", "min_duration")) {
```
Comment (mxpv) [inline services/update/matcher.go:57]: Do we need a separate helper for this? This looks simple enough:

```go
if filters.MaxDuration > 0 && filters.MaxDuration > episode.Duration {
    logger.Infof("skipping due to duration filter (episode: %d, max duration: %d)", ...)
    return false
}

if filters.MinDuration > 0 && filter.MinDuration < episode.Duration {
    ...
    return false
}
```

**Comment by mxpv** [comment]
Also needs rebase

**Comment by Th0masL** [comment]
I've removed the helpers, as suggested :)

Here is the overview of the log file, in debug mode:

```bash
DEBU[2022-11-23T20:07:49+02:00] -> SomeYoutubeChannel (update '@every 4h0m0s')
INFO[2022-11-23T20:07:49+02:00] -> updating https://www.youtube.com/user/SomeYoutubeChannel  feed_id=SomeYoutubeChannel format=audio quality=low
DEBU[2022-11-23T20:07:49+02:00] building feed
DEBU[2022-11-23T20:07:49+02:00] Expected to make 1 API calls to get the descriptions for 3 episode(s).
DEBU[2022-11-23T20:07:49+02:00] received 3 episode(s) for "SomeYoutubeChannel"
DEBU[2022-11-23T20:07:49+02:00] successfully saved updates to storage
INFO[2022-11-23T20:07:49+02:00] downloading episodes                          page_size=3
INFO[2022-11-23T20:07:49+02:00] skipping due to regexp mismatch               episode_id=QSUyW2SYxyz filter=title
INFO[2022-11-23T20:07:49+02:00] skipping due to duration filter (1616s)       episode_id=Q5a81JNaxyz filter=min_duration
DEBU[2022-11-23T20:07:49+02:00] adding f6Vkhov_xyz ("Some video name") to queue
INFO[2022-11-23T20:07:49+02:00] download count: 1
INFO[2022-11-23T20:07:49+02:00] ! downloading episode https://youtube.com/watch?v=f6Vkhov_xyz  episode_id=f6Vkhov_xyz index=0
DEBU[2022-11-23T20:08:50+02:00] copying file                                  episode_id=f6Vkhov_xyz index=0
INFO[2022-11-23T20:08:50+02:00] creating file: /tmp/podsync/data/SomeYoutubeChannel/f6Vkhov_xyz.mp3  name=SomeYoutubeChannel/f6Vkhov_xyz.mp3
DEBU[2022-11-23T20:08:50+02:00] written 71075789 bytes                        name=SomeYoutubeChannel/f6Vkhov_xyz.mp3
INFO[2022-11-23T20:08:50+02:00] successfully downloaded file "f6Vkhov_xyz"    episode_id=f6Vkhov_xyz index=0
INFO[2022-11-23T20:08:50+02:00] downloaded 1 episode(s)
INFO[2022-11-23T20:08:50+02:00] nothing to clean                              feed_id=SomeYoutubeChannel
DEBU[2022-11-23T20:08:50+02:00] building iTunes podcast feed
INFO[2022-11-23T20:08:50+02:00] creating file: /tmp/podsync/data/SomeYoutubeChannel.xml  name=SomeYoutubeChannel.xml
DEBU[2022-11-23T20:08:50+02:00] written 10517 bytes                           name=SomeYoutubeChannel.xml
DEBU[2022-11-23T20:08:50+02:00] building podcast OPML
INFO[2022-11-23T20:08:50+02:00] creating file: /tmp/podsync/data/podsync.opml  name=podsync.opml
DEBU[2022-11-23T20:08:50+02:00] written 130 bytes                             name=podsync.opml
INFO[2022-11-23T20:08:50+02:00] successfully updated feed in 1m0.980123362s
INFO[2022-11-23T20:08:50+02:00] next update of SomeYoutubeChannel: 2022-11-24 00:07:49 +0200 EET
```

**Review by mxpv — APPROVED** [review]
Thanks!

**Pull request merged**